client/client.go

```diff
@@ -302,12 +302,28 @@ func addChange(cl *changelist.FileChangelist, c changelist.Change, roles ...stri
 	return nil
 }
 
-// AddDelegation creates a new changelist entry to add a delegation to the repository
-// when the changelist gets applied at publish time.  This does not do any validation
-// other than checking the name of the delegation to add - all that will happen
-// at publish time.
-func (r *NotaryRepository) AddDelegation(name string, threshold int,
-	delegationKeys []data.PublicKey, paths []string) error {
+// AddDelegation creates changelist entries to add provided delegation public keys and paths.
+// This method composes AddDelegationRoleAndKeys and AddDelegationPaths (each creates one changelist if called).
+func (r *NotaryRepository) AddDelegation(name string, delegationKeys []data.PublicKey, paths []string) error {
```

Comment:
Normally I'd say it'd be good to future proof this, since thresholds are in the spec, but we don't have support for changing any other thresholds, nor do we know (nor does the spec seem to suggest) a good way for managing multiple signers.  So I don't feel particularly strongly.

If and when we support thresholds, we could add a "ChangeThreshold/UpdateKeys" function that takes any role maybe?  To that end, if we did want to futureproof, we can consider allowing adding keys for any role.  (I'm not saying we should do this now)